qcodes/parameters/parameter_base.py

```diff
@@ -830,6 +834,15 @@ def full_name(self) -> str:
         """
         return "_".join(self.name_parts)
 
+    @property
+    def register_name(self) -> str:
+        """
+        Name that will be used to register this parameter in a dataset
+        By default, this returns the full_name, but may be overriden if custom
+        behavior is desired.
```

Comment:
```suggestion
        By default, this returns the ``full_name``, or the value of
        ``register_name`` argument if it was passed at inititalization of the parameter.
        This property may be overriden if custom behavior is desired.
```

Reply:
On second thought, there's no need to include the last sentence, as all python properties can be overriden to produce custom behavior.